### ISSUE TITLE
error: llvm function @abs and ar intrinsic @ar.libc.abs have a different type

### ISSUE THREAD

**Issue by practicalswift** [NASA-SW-VnV/ikos#53]
```
$ cat testcase.cpp
extern "C" void abs();
void a() { abs(); }
$ ikos testcase.cpp
[*] Compiling testcase.cpp
[*] Running ikos preprocessor
[*] Running ikos analyzer
[*] Translating LLVM bitcode to AR
ikos-analyzer: /tmp/ikos-OceEsM/testcase.bc: error: llvm function @abs and ar intrinsic @ar.libc.abs have a different type
ikos: error: a run-time error occurred
```

**Comment by arthaud** [comment]
Hi @practicalswift,

Since `abs` is a function defined in the C standard library, IKOS expects it to be defined as:
```cpp
extern "C" int abs(int);
```

If you do not want to use the C standard library, you can provide the option `--no-libc` to IKOS.

Since this seems like a legitimate error to me, I'm proposing to close this issue.

PS: If you think we could improve the error message, feel free to suggest a better alternative ;)

**Comment by practicalswift** [comment]
Yes, the test case is a bit crazy. A better error message instead of the run-time error is probably enough to solve this.

What about something along the lines of "The function definition does not match the expected C standard library definition."? :-)

**Comment by practicalswift** [comment]
Confirming that the fix works as expected. Now getting a user-friendly error message :-)
